cabal-install/Distribution/Client/ProjectPlanOutput.hs

```diff
@@ -155,7 +155,9 @@ encodePlanAsJson distDirLayout elaboratedInstallPlan elaboratedSharedConfig =
             let components = J.object $
                   [ comp2str c J..= (J.object $
                     [ "depends"     J..= map (jdisplay . confInstId) ldeps
-                    , "exe-depends" J..= map (jdisplay . confInstId) edeps ] ++
+                    , "exe-depends" J..= map (jdisplay . confInstId) edeps
+                    , "requested-by-default" J..= requestedByDefaultJ c ] ++
```

Comment:
Since most of the time this will be redundantly `requested-by-default: True` and I intend `plan.json` to be manually optimized to reduce redundancy and noise, I'd rather this was something like

```suggestion
                    ] ++ [ "optional" J..= True | not (requestedByDefaultJ c) ] ++
```